apps/src/javalab/ControlButtons.jsx

```diff
@@ -4,76 +4,111 @@ import i18n from '@cdo/locale';
-        onClick={toggleTest}
-        isHorizontal
-        className="noBorder"
-        style={buttonStyles}
-      />
+    <div>
+      <div style={styles.leftButtons}>
+        <JavalabButton
+          text={isRunning ? i18n.stop() : i18n.runProgram()}
+          icon={
+            <FontAwesome icon={isRunning ? 'stop' : 'play'} className="fa" />
+          }
+          onClick={toggleRun}
+          isHorizontal
+          style={{...styles.button.all, ...styles.button.orange}}
+        />
+        <JavalabButton
+          text={isTesting ? i18n.stopTests() : i18n.test()}
+          icon={<FontAwesome icon="flask" className="fa" />}
+          onClick={toggleTest}
+          isHorizontal
+          style={{...styles.button.all, ...styles.button.white}}
+        />
+      </div>
+      <div style={styles.rightButtons}>
+        <JavalabSettings>{renderSettings()}</JavalabSettings>
+        {!isEditingStartSources && (
+          <JavalabButton
+            text={i18n.finish()}
+            onClick={onContinue}
+            style={{...styles.button.all, ...styles.button.blue}}
+            isDisabled={isReadOnlyWorkspace}
+          />
+        )}
+      </div>
     </div>
   );
 }
 
 ControlButtons.propTypes = {
-  isDarkMode: PropTypes.bool.isRequired,
   isRunning: PropTypes.bool.isRequired,
   isTesting: PropTypes.bool.isRequired,
   toggleRun: PropTypes.func.isRequired,
-  toggleTest: PropTypes.func.isRequired
+  toggleTest: PropTypes.func.isRequired,
+  isEditingStartSources: PropTypes.bool,
+  isReadOnlyWorkspace: PropTypes.bool.isRequired,
+  onContinue: PropTypes.func.isRequired,
+  renderSettings: PropTypes.func.isRequired
 };
 
 const styles = {
-  container: {
-    display: 'flex',
-    flexDirection: 'column',
-    justifyContent: 'space-around'
+  leftButtons: {
+    float: 'left'
+  },
+  rightButtons: {
+    float: 'right'
   },
   button: {
     all: {
-      fontSize: 16,
+      float: 'left',
+      fontSize: 15,
       width: 140,
-      backgroundColor: 'transparent',
+      backgroundColor: color.orange,
+      borderColor: color.orange,
+      fontFamily: '"Gotham 5r"',
+      padding: '5px 12px',
+      margin: '5px 0 5px 5px',
+      justifyContent: 'center',
+      ':hover': {
+        color: color.white,
+        boxShadow: 'none'
+      }
+    },
+    orange: {
+      backgroundColor: color.orange,
+      borderColor: color.orange
+    },
+    white: {
+      backgroundColor: color.white,
+      borderColor: color.dark_charcoal,
+      color: color.dark_charcoal,
       ':hover': {
-        color: color.orange,
+        color: color.dark_charcoal,
         boxShadow: 'none'
       }
     },
-    light: {color: color.cyan},
-    dark: {color: color.lightest_gray}
+    blue: {
+      backgroundColor: color.default_blue,
+      borderColor: color.default_blue
+    }
+  },
+  finish: {
+    backgroundColor: color.orange,
+    borderColor: color.orange,
+    fontFamily: '"Gotham 5r"',
+    fontSize: '15px',
```

Comment:
Just curious - do we have a preference for using or not using constants for font families and font sizes as we do with colors?

Reply:
Good catch, and I think we should do this.  Added to backlog in the interests of getting this big change through.